torch/csrc/autograd/variable.h

```diff
@@ -189,6 +189,9 @@ struct TORCH_API AutogradMeta : public c10::AutogradMetaInterface {
   // Only meaningful on leaf variables (must be false otherwise)
   bool requires_grad_;
 
+  // Only meaningful on non-leaf variables (must be false otherwise)
+  bool retains_grad_ = false;
```

Comment:
nit: to be consistent with the other variables here, maybe you want to do this initialization in the constructor?

Reply:
Thanks for the catch! Yes it's better to put this in the constructor :D